### DISCUSSION TITLE
set and get nested values

### DISCUSSION THREAD

**Discussion by hui21109** [react-hook-form/react-hook-form#2249] [Q&A]
I have a component might have two onChange functions
For a special purpose. My data is an array-like data: [‘aaa’, ‘bbb’]
how can I update each index of the data separately.

example https://codesandbox.io/s/react-hook-form-v6-controller-9ejd9?file=/src/index.jsx

**Comment by bluebill1049** [comment]
why are you wrapped with Controller? it should probably just be two input and update separately.

**Reply by hui21109** [reply]
Thanks for the quick reply!

What if there is a third party component and we need to do so ? 
so, is there any possibility I can update nested data separately?
Or it is just because I wrapped it with Controller so it won't work?

**Reply by bluebill1049** [reply]
can you provide a thrid part component which need to do that in a codesandbox?

**Reply by hui21109** [reply]
actually it is a component we will avoid to change it because of the effort is too large.
I find a solution
` 
<CusomInput
  onChangeText1={e => onChange([e.target.value, value[1]])}
  text1={value[0]}
  onChangeText2={e => onChange([value[0], e.target.value])}
  text2={value[1]} // suppose to be value[1]?
 />
`
Any suggestion?
